### PR TITLE
[WIP]: feat: support creating multiple widget adapter from one DOM elements.

### PR DESCRIPTION
**Pull request by andycall**
In earlier versions of WebF, an HTMLElement could only have one adapter in the Flutter widget system at a time. When a new widget adapter was created for an HTMLElement, the previously created adapter's RenderObjects would be removed and disposed of.

For certain use cases, such as ContextMenu Flutter widgets, multiple widgets were required to function properly, which conflicted with the initial design of our adapter system.

In this PR, we introduce a new adapter system that creates separate RenderObjects for each widget while sharing the same RenderStyle and DOM element instance.

This means that if we apply styles or add new DOM nodes to the HTMLElement associated with multiple adapters, all the RenderObjects created by these adapters will be updated, resulting in consistent rendering. Additionally, when one of these adapters handles a new event, the event callbacks on the DOM element will be triggered accordingly.

image

### CODE REVIEW

File: webf/lib/src/widget/webf_adapter_widget.dart [hunk 1]
```diff
@@ -66,49 +65,20 @@ class WebFWidgetElementState extends State<WebFWidgetElementStatefulWidget> {
...
-  List<Widget>? get children => _cachedChildren;
-
-  void markChildrenNeedsUpdate() {
-    _cachedChildren = null;
-    if (mounted) {
-      requestUpdateState();
-    }
-  }
-
   Widget buildNodeWidget(dom.Node node, {Key? key}) {
```
Comment (looseyi) [inline webf/lib/src/widget/webf_adapter_widget.dart:68]: this is uesless ?

Comment (andycall) [reply webf/lib/src/widget/webf_adapter_widget.dart:68]: yep

File: webf/lib/src/dom/child_node_list.dart [hunk 2]
```diff
@@ -53,6 +54,10 @@ class ChildNodeList extends NodeList {
     return _collectionIndexCache.nodeAt(this, index);
   }
 
+  List<Widget> toWidgetList() {
+    return map((node) => node.toWidget()).toList();
```
Comment (looseyi) [inline webf/lib/src/dom/child_node_list.dart:58]: should filter Comment element here ?
```
    return where((node) => node is! Comment).map((node) => node.toWidget()).toList();
```

**Comment by andycall** [comment]
migrated to enterprise

**Pull request closed without merging**